src/Compilers/CSharp/Portable/Binder/BindingDiagnosticBag.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal static class BindingDiagnosticBagExtensions
+    {
+        internal static CSDiagnosticInfo Add(this CodeAnalysis.BindingDiagnosticBag diagnostics, ErrorCode code, Location location)
+        {
+            var info = new CSDiagnosticInfo(code);
+            diagnostics.DiagnosticBag?.Add(new CSDiagnostic(info, location));
+            return info;
+        }
+
+        internal static CSDiagnosticInfo Add(this CodeAnalysis.BindingDiagnosticBag diagnostics, ErrorCode code, Location location, params object[] args)
+        {
+            var info = new CSDiagnosticInfo(code, args);
+            diagnostics.DiagnosticBag?.Add(new CSDiagnostic(info, location));
+            return info;
+        }
+
+        internal static CSDiagnosticInfo Add(this CodeAnalysis.BindingDiagnosticBag diagnostics, ErrorCode code, Location location, ImmutableArray<Symbol> symbols, params object[] args)
+        {
+            var info = new CSDiagnosticInfo(code, args, symbols, ImmutableArray<global::Microsoft.CodeAnalysis.Location>.Empty);
+            diagnostics.DiagnosticBag?.Add(new CSDiagnostic(info, location));
+            return info;
+        }
+
+        internal static void Add(this CodeAnalysis.BindingDiagnosticBag diagnostics, DiagnosticInfo? info, Location location)
+        {
+            if (info is object)
+            {
+                diagnostics.DiagnosticBag?.Add(info, location);
+            }
+        }
+
+        internal static void AddForSymbol(this ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo, Symbol? symbol, bool addDiagnostics = true, bool addDependencies = true)
+        {
+            if (symbol is null || useSiteInfo.IsDiscarded || !(addDiagnostics || addDependencies))
```

Comment:
>|| !(addDiagnostics || addDependencies) [](start = 58, length = 39)

I think this reads better as `&& !addDiagnostics && !addDependencies`

Reply:
> I think this reads better as ```&& !addDiagnostics && !addDependencies```

Perhaps, but that wouldn't do the same thing.

---
In reply to: [361554171](https://github.com/dotnet/roslyn/pull/40560#discussion_r361554171) [](ancestors = 361554171)

Reply:
> Perhaps, but that wouldn't do the same thing.

I edited

---
In reply to: [361554390](https://github.com/dotnet/roslyn/pull/40560#discussion_r361554390) [](ancestors = 361554390,361554171)